modules/core/src/test/java/org/apache/ignite/testframework/junits/logger/GridTestLog4jLogger.java

```
@@ -466,6 +468,8 @@ public static void removeAppender(FileAppender a) {
         if (!impl.isDebugEnabled())
             warning("Logging at DEBUG level without checking if DEBUG level is enabled: " + msg);
 
+        assert impl.isDebugEnabled() : "Logging at DEBUG level without checking if DEBUG level is enabled: " + msg;
```

I think you should not throw an exception

---

Why?

---

If DEBUG is not enabled, you should throw an error so that when you disable the "ea" option, you do not miss the error.

---

"-ea" is enabled on TeamCity runs and IDE test configuration. Anyway according to your logic user can set other logger implementation and avoid those checks.